modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/internal/upgrade/BlogsServiceUpgrade.java

```diff
@@ -95,12 +101,55 @@ public void register(Registry registry) {
 			new com.liferay.blogs.internal.upgrade.v2_1_1.UpgradeBlogsEntry());
 	}
 
+	private UnsafeFunction<String, Boolean, Exception>
+		_getUpgradeDiscussionSubscriptionClassNameUnsafeFunction() {
+
+		return className -> {
+			DynamicQuery groupDynamicQuery = _groupLocalService.dynamicQuery();
+
+			groupDynamicQuery.add(RestrictionsFactoryUtil.eq("site", true));
+
+			List<Long> groupIds = ListUtil.toList(
+				_groupLocalService.dynamicQuery(groupDynamicQuery),
+				Group.GROUP_ID_ACCESSOR);
+
+			DynamicQuery subscriptionDynamicQuery =
+				_subscriptionLocalService.dynamicQuery();
+
+			subscriptionDynamicQuery.add(
+				RestrictionsFactoryUtil.eq(
+					"classNameId",
+					_classNameLocalService.getClassNameId(className)));
+
+			List<Subscription> subscriptions =
+				_subscriptionLocalService.dynamicQuery(
+					subscriptionDynamicQuery);
+
+			for (Subscription subscription : subscriptions) {
+				if (groupIds.contains(subscription.getClassPK())) {
```

Comment:
@robertoDiaz @IstvanD 

Is this check required? Shouldn't all subscriptions associated with the Group className be related to Groups?

Reply:
Subscriptions are related with the Blogs classname, not with the Groups.

In blogs we have 2 subscriptions patterns className - classPK:

1. Blogs.class - groupId (suscribed to all the changes in the blogs portlet)
2. Blogs.class - blogsEntryId (susbcribed to the comments of a single blogs entry)

What we are doing here is filtering and modifying only the second ones (adding the Discussion preffix)